bucketstracking_agentmanager.go

```diff
@@ -263,25 +279,52 @@ func (m *BucketsTrackingAgentManager) makeAgent(ctx context.Context, bucketName
 	})
 }
 
+func (m *BucketsTrackingAgentManager) Watch(ctx context.Context) <-chan *ParsedConfig {
```

Comment:
I think this should actually be `WatchConfig` since the 'Config' portion is not implied when talking about a component that is not config oriented (like `BucketsTrackingAgentManager`).  Additionally, is there some reason we couldn't just use the underlying topology provider directly?
```
func (m *BucketsTrackingAgentManager) WatchConfig(ctx context.Context) <-chan *ParsedConfig {
  return m.topologyCfgWatcher.Watch(ctx)
}
```

Reply:
The underlying watcher is just a watcher, it doesn't filter what it pushes out based on rev id or config validity. I was thinking that by only pushing out configs applied by the manager then it dedupes that logic being scattered everywhere.

Reply:
What logic is it that you're referring to?  I would expect that any deduplication logic and what not would already be getting applied by the topology watcher itself, since it's meant to be the centralized component that provides the configuration updates.

Reply:
I was going to put it in the topology watcher but the topology watcher isn't actually a component, it's just a config watcher http at present. I'd feel better about creating a topology watcher which handles the deduplication logic.